opengrok-indexer/src/main/java/org/opengrok/indexer/web/SearchHelper.java

```diff
@@ -814,4 +814,61 @@ private void ensureSettingsHelper() {
             settingsHelper = new SettingsHelper(reader);
         }
     }
+
+    /**
+     * Search Helper Builder Class.
+     */
+    public static final class Builder {
+        private final SearchHelper searchHelper;
+
+        /**
+         * Search Helper Builder Class constructor.
+         * start parameter is defaulted to zero
+         * maxItems parameter is defaulted to 10
+         * all boolean parameters(crossRefSearch,guiSearch,noRedirect) is defaulted to false
+         * order is defaulted to RELEVANCY based sorting
+         * @param dataRoot dataRoot
+         * @param sourceRoot sourceRoot
+         * @param eftarFileReader eftarFileReader
+         * @param queryBuilder queryBuilder
+         * @param contextPath contextPath
+         */
+        public Builder(File dataRoot, File sourceRoot,
+                     EftarFileReader eftarFileReader, QueryBuilder queryBuilder,
+                     String contextPath) {
+            searchHelper = new SearchHelper(dataRoot, sourceRoot, eftarFileReader, queryBuilder, contextPath);
+        }
+        public Builder start(int start) {
+            searchHelper.start = start;
+            return this;
+        }
```

Comment:
nit: insert spaces between the method definitions to be consistent with the methods below